File: algorithms/34. Search for a Range/solution.cc
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        auto range = equal_range(nums.begin(), nums.end(), target);
        if ( range.first == range.second ) return {-1, -1};
        return {range.first - nums.begin(), range.second - nums.begin() - 1};
    }
};

File: algorithms/34. Search for a Range/solution.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        lp = bisect.bisect_left(nums, target)
        rp = bisect.bisect_right(nums, target)
        return [-1, -1] if target not in nums[lp:lp+1] else [lp, rp-1]
